### PR TITLE
Add secretlint

### PR DESCRIPTION
**Pull request by chapati23**
### Description
This PR adds [secretlint](https://github.com/secretlint/secretlint) to our toolchain.

- [x] Added secretlint and a few plugins
- [x] Added `npm run lint:secrets` task and included it into `npm run lint` task
- [x] Enforce running of secretlint in our `pre-commit` hook

image

### How to test

1. Check out this branch locally
2. Try it out with `npm run lint:secrets`, you should see errors because we accidentally committed secrets into our env.example files
3. Try committing something (can be an empty commit), and check that it won't let you because secretlint is failing

### CODE REVIEW

File: .secretlintrc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "rules": [
+    { "id": "@secretlint/secretlint-rule-no-k8s-kind-secret" },
```
Comment (bayological) [inline .secretlintrc.json:3]: For this repo, this rule is not needed.

**Review by bayological — APPROVED** [review]
Nice addition to this repo. We should consider adding it to our other repos.

**Comment by RyRy79261** [comment]
@chapati23 Looks good, though with it throwing errors, would that not mean the deployments will fail with these files? I'd assume this lint would have no errors before this PR can merge

**Review by RyRy79261 — APPROVED** [review]

File: .secretlintrc.json (hunk 1 above)
Comment (chapati23) [reply .secretlintrc.json]: yeah good catch, was trying out the different plugins and forgot to remove this again, gone now

**Pull request merged**